core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3049,16 +3049,14 @@ public PhysicalOperation visitUpdate(UpdateNode node, LocalExecutionPlanContext
         {
             Integer[] columnValueAndRowIdChannels = new Integer[columnValueAndRowIdSymbols.size()];
             int symbolCounter = 0;
-            // This depends on the outputSymbols being ordered as the blocks of the
-            // resulting page are ordered.
-            for (Symbol symbol : outputSymbols) {
-                int index = columnValueAndRowIdSymbols.indexOf(symbol);
+            for (Symbol symbol : columnValueAndRowIdSymbols) {
+                int index = outputSymbols.indexOf(symbol);
                 if (index >= 0) {
```

Comment:
It feels to me that we should have `verify(index >= 0)` here. Is that valid case that there exists a symbol in `columnValueAndRowIdSymbols` which is not part of `outputSymbols`?

Reply:
Good point - - added
`verify(index >= 0, "Could not find symbol %s in the outputSymbols %s", symbol, outputSymbols);`